### PR TITLE
Backend Plugins: Refactor backend plugin registration and start

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Moves the details of loading plugins into the backend
plugin manager from the respective plugin (datasource,
transform and renderer).

### CODE REVIEW

File: pkg/plugins/transform_plugin.go [hunk 1]
```diff
@@ -28,45 +28,38 @@ type TransformPlugin struct {
 	*TransformWrapper
 }
 
-func (tp *TransformPlugin) Load(decoder *json.Decoder, pluginDir string) error {
-	if err := decoder.Decode(&tp); err != nil {
+func (p *TransformPlugin) Load(decoder *json.Decoder, pluginDir string) error {
+	if err := decoder.Decode(&p); err != nil {
```
Comment (aknuds1) [inline pkg/plugins/transform_plugin.go]: I think we don't need a pointer to a pointer here? I tested the below and it compiles at least:
```suggestion
	if err := decoder.Decode(p); err != nil {
```

I know it's not directly related to the changes here, but we might as well clean a bit.

**Review by aknuds1 — DISMISSED** [review]
Generally looks good to me, but I think we can remove the `execute` wrappers.

File: pkg/plugins/backendplugin/client.go [hunk 2]
```diff
@@ -41,17 +41,42 @@ func newClientConfig(executablePath string, logger log.Logger, versionedPlugins
 	}
 }
 
+// LegacyCallbackFunc callback function called when a plugin with old plugin protocol is started.
```
Comment (kylebrandt) [inline pkg/plugins/backendplugin/client.go]: I'm not sure if "callback" is best here. If it is a once every start thing (and when restarted?). The maybe InitFuncs or something?

File: pkg/plugins/backendplugin/client.go [hunk 3]
```diff
@@ -41,17 +41,42 @@ func newClientConfig(executablePath string, logger log.Logger, versionedPlugins
...
+type LegacyCallbackFunc func(pluginID string, client *LegacyClient, logger log.Logger) error
+
+func (f LegacyCallbackFunc) execute(pluginID string, client *LegacyClient, logger log.Logger) error {
```
[the remark this answers is not in the document]
Comment (marefr) [reply pkg/plugins/backendplugin/client.go]: Probably not, it did made the intellisense work better. But can remove it.

**Review by kylebrandt — APPROVED** [review]

**Comment by torkelo** [comment]
Really happy this part of the backend is getting some refactoring and more design/reuse :heart:

File: pkg/plugins/backendplugin/client.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (aknuds1) [reply pkg/plugins/backendplugin/client.go]: @marefr Thanks!

**Review by aknuds1 — COMMENTED** [review]
LGTM!

**Pull request merged**